mesheryctl/internal/cli/root/perf/result.go

```diff
@@ -0,0 +1,240 @@
+	MesheryID     *uuid.UUID
+	LoadGenerator string
+}
+
+var (
+	resultPage int
+)
+
+var resultCmd = &cobra.Command{
+	Use:   "result profile-name",
+	Short: "List performance test results",
+	Long:  `List all the available test results of a performance profile`,
+	Args:  cobra.MinimumNArgs(1),
+	Example: `
+// List Test results (maximum 25 results)	
+mesheryctl perf result saturday profile 
+
+// View performance results with more information (maximum 25 results)
+mesheryctl perf result saturday profile --expand
+
+// View other set of performance results with --page (maximum 25 results)
+mesheryctl perf result saturday profile --page 2
+`,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		// used for searching performance profile
+		var searchString, profileID string
+		var profileResp *models.PerformanceProfilesAPIResponse
+		// setting up for error formatting
+		cmdUsed = "result"
+
+		mctlCfg, err := config.GetMesheryCtl(viper.GetViper())
+		if err != nil {
+			return ErrMesheryConfig(err)
+		}
+
+		profileURL := mctlCfg.GetBaseMesheryURL() + "/api/user/performance/profiles"
+
+		// set default tokenpath for command.
+		if tokenPath == "" {
+			tokenPath = constants.GetCurrentAuthToken()
+		}
+
+		// Merge args to get result-name
+		searchString = strings.Join(args, "%20")
+
+		_, _, body, err := fetchPerformanceProfiles(profileURL, searchString)
+		if err != nil {
+			return err
+		}
+		err = json.Unmarshal(body, &profileResp)
+		if err != nil {
+			return err
+		}
+
+		if len(profileResp.Profiles) == 0 {
+			log.Info("No Performance Profiles found with the given name")
+			return nil
+		}
+
+		if len(profileResp.Profiles) == 1 {
+			// found only one profile with matching name
+			profileID = profileResp.Profiles[0].ID.String()
+		} else {
+			fmt.Printf("Found multiple profiles with given name\n")
+			fmt.Printf("Select a performance profile\n")
+			fmt.Printf("-----------------------------\n")
+			selectedProfileIndex := multipleProfileConfirmation(profileResp.Profiles)
+			profileID = profileResp.Profiles[selectedProfileIndex].ID.String()
+			fmt.Printf("\n")
+		}
+
+		resultURL := mctlCfg.GetBaseMesheryURL() + "/api/user/performance/profiles/" + profileID + "/results"
+
+		data, expandedData, body, err := fetchPerformanceProfileResults(resultURL)
+		if err != nil {
+			return err
+		}
+
+		if len(data) == 0 {
+			log.Info("No Test Results to display")
+			return nil
+		}
+
+		if outputFormatFlag != "" {
+			var tempStruct *models.PerformanceResultsAPIResponse
+			err = json.Unmarshal(body, &tempStruct)
+			if err != nil {
+				return ErrFailUnmarshal(err)
+			}
+			body, _ = json.Marshal(tempStruct.Results)
+			if outputFormatFlag == "yaml" {
+				body, _ = yaml.JSONToYAML(body)
+			} else if outputFormatFlag != "json" {
+				return ErrInvalidOutputChoice()
+			}
+			log.Info(string(body))
+		} else if !expand {
+			utils.PrintToTable([]string{"NAME", "MESH", "QPS", "DURATION", "P50", "P99.9", "START-TIME"}, data)
```

Comment:
We do not have an option to view the result of a single test right?

Was that the way we wanted this?

Reply:
Could we do `perf result <profile name> <result name>`?

Or `perf result <profile name> view <result name>`?

Anyway, we need this function as well.

Reply:
This is a good idea, but test result names are going to be removed from the system officially. and the generated names also match up with each other and if someone again will search some result, I have to end-up printing all of them.